core/command_line_parser.h

```diff
@@ -0,0 +1,256 @@
+/* Copyright (c) 2014-2021 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef COMMAND_LINE_PARSER_H
+#define COMMAND_LINE_PARSER_H
+
+#include "core/object/reference.h"
+#include "core/templates/ordered_hash_map.h"
+
+class CommandLineOption : public Reference {
+	GDCLASS(CommandLineOption, Reference);
+
+	struct ArgumentChecker;
+	struct FunctionChecker;
+	struct CallableChecker;
+
+	// Names for the options. e.g --help or -h.
+	PackedStringArray _names;
+	// List of default values for each argument, empty if any value is allowed.
+	PackedStringArray _default_args;
+	// List of allowed values for each argument, empty if any value is allowed.
+	PackedStringArray _allowed_args;
+	// Option description that will be displayed in help.
+	String _description;
+	// Option category, options sharing the same category are grouped together in the help text.
+	String _category;
+	// Name for the option arguments that will be displayed in help text.
+	String _arg_text = RTR("<arg>");
+	// Make the option visible in help.
+	bool _hidden = false;
+	// If true, arguments can be specified without option name.
+	bool _positional = false;
+	// If true, argument always should be provided.
+	bool _required = false;
+	// If true, the option can be specified several times.
+	bool _multitoken = false;
+	// Arguments count required for the option, -1 for all arguments left.
+	int _args_count = 1;
```

Comment:
Recall `get_options_count` → `get_option_count` suggestion? This one should also likely use singular form.

```suggestion
	int _arg_count = 1;
```

If you grep for `arg_count` throughout Godot's codebase, you'll know what I mean, so this change should make it consistent. Also consistent with `arg_text`, although slightly unrelated.